examples/tv-app/linux/include/channel/ChannelManager.cpp

```diff
@@ -16,76 +16,158 @@
  */
 
 #include "ChannelManager.h"
+#include <vector>
 
 using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters::Channel;
 
+ChannelManager::ChannelManager()
+{
+    ChannelInfoType abc;
+    abc.affiliateCallSign = MakeOptional(chip::CharSpan::fromCharString("KAAL"));
+    abc.callSign          = MakeOptional(chip::CharSpan::fromCharString("KAAL-TV"));
+    abc.name              = MakeOptional(chip::CharSpan::fromCharString("ABC"));
+    abc.majorNumber       = static_cast<uint8_t>(6);
+    abc.minorNumber       = static_cast<uint16_t>(0);
+    mChannels.push_back(abc);
+
+    ChannelInfoType pbs;
+    pbs.affiliateCallSign = MakeOptional(chip::CharSpan::fromCharString("KCTS"));
+    pbs.callSign          = MakeOptional(chip::CharSpan::fromCharString("KCTS-TV"));
+    pbs.name              = MakeOptional(chip::CharSpan::fromCharString("PBS"));
+    pbs.majorNumber       = static_cast<uint8_t>(9);
+    pbs.minorNumber       = static_cast<uint16_t>(1);
+    mChannels.push_back(pbs);
+
+    ChannelInfoType pbsKids;
+    pbsKids.affiliateCallSign = MakeOptional(chip::CharSpan::fromCharString("KCTS"));
+    pbsKids.callSign          = MakeOptional(chip::CharSpan::fromCharString("KCTS-TV"));
+    pbsKids.name              = MakeOptional(chip::CharSpan::fromCharString("PBS Kids"));
+    pbsKids.majorNumber       = static_cast<uint8_t>(9);
+    pbsKids.minorNumber       = static_cast<uint16_t>(2);
+    mChannels.push_back(pbsKids);
+
+    ChannelInfoType worldChannel;
+    worldChannel.affiliateCallSign = MakeOptional(chip::CharSpan::fromCharString("KCTS"));
+    worldChannel.callSign          = MakeOptional(chip::CharSpan::fromCharString("KCTS-TV"));
+    worldChannel.name              = MakeOptional(chip::CharSpan::fromCharString("World Channel"));
+    worldChannel.majorNumber       = static_cast<uint8_t>(9);
+    worldChannel.minorNumber       = static_cast<uint16_t>(3);
+    mChannels.push_back(worldChannel);
+
+    mCurrentChannelIndex = 0;
+    mCurrentChannel      = mChannels[mCurrentChannelIndex];
+}
+
 CHIP_ERROR ChannelManager::HandleGetChannelList(AttributeValueEncoder & aEncoder)
 {
     // TODO: Insert code here
-    return aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
-        int maximumVectorSize = 2;
-
-        for (int i = 0; i < maximumVectorSize; ++i)
+    std::vector<ChannelInfoType> channels = mChannels;
```

Comment:
Why do we need to make a copy?  Why can't we just close over `mChannels` by reference or `this` by value?

Reply:
Updated.